pkgs/servers/draupnir/default.nix

```diff
@@ -0,0 +1,98 @@
+{ lib
+, nixosTests
+, stdenv
+, fetchFromGitHub
+, makeWrapper
+, nodejs
+, pkgs
+, mkYarnPackage
+, fetchYarnDeps
+}:
+
+# docs: https://github.com/NixOS/nixpkgs/blob/master/doc/languages-frameworks/javascript.section.md#yarn2nix-javascript-yarn2nix
+
+mkYarnPackage rec {
+  pname = "draupnir";
+  version = "1.81.1";
+  # NOTE: Hard fork of Mjolnir.
+  src = fetchFromGitHub {
+    owner = "Gnuxie";
+    repo = "Draupnir";
+    rev = "v${version}";
+    sha256 = "RyVTdGnt4LbOyqYgSxbNpCYIFeYxgvlkDlFWN1Jw4DY=";
+  };
+
+  nativeBuildInputs = [
+    nodejs
+    makeWrapper
+    pkgs.yarn
+    pkgs.matrix-sdk-crypto-nodejs
+  ];
+  offlineCache = fetchYarnDeps {
+    yarnLock = src + "/yarn.lock";
+    hash = "sha256-wXlaYC2RPPTSwCkuw3TsRAMV5sKsZrJDdanj16p1CXA=";
+  };
+  yarnLock = src + "/yarn.lock";
+  packageJSON = src + "/package.json";
+
+  buildPhase = ''
+      echo "Running preBuild"
+      runHook preBuild
+      export HOME=$(mktemp -d)
+      echo "Building..."
+      yarn --offline --verbose build
+      # copy built modules to package...
+      echo "Copying built matrix-sdk-crypto-nodejs modules to package..."
+      cp -a ${pkgs.matrix-sdk-crypto-nodejs}/lib/node_modules/* node_modules/
+      echo "Adding version.txt..."
+      mkdir -p deps/draupnir/
+      echo "${version}-nix" > deps/draupnir/version.txt
+      echo "Running postBuild..."
+      runHook postBuild
+    '';
+
+  installPhase = ''
+    runHook preInstall
+
+    ls /build/source
+    mkdir -p $out/share
+    cp -a . $out/share/draupnir
+
+    #include config
+    cp $src/config $out/share/draupnir/ -rv
+
+    makeWrapper ${nodejs}/bin/node $out/bin/draupnir \
+      --add-flags $out/share/draupnir/deps/draupnir/lib/index.js
+
+    runHook postInstall
+  '';
+  distPhase = "true";
+
+  passthru = {
+    #tests = {
+    #  inherit (nixosTests) draupnir;
+    #};
+    updateScript = ./update.sh;
+  };
+
+  meta = with lib; {
+    description = "A moderation tool for Matrix";
+    homepage = "https://github.com/Gnuxie/Draupnir";
+    longDescription = ''
+      As an all-in-one moderation tool, it can protect your server from
+      malicious invites, spam messages, and whatever else you don't want.
+      In addition to server-level protection, Draupnir is great for communities
+      wanting to protect their rooms without having to use their personal
+      accounts for moderation.
+
+      The bot by default includes support for bans, redactions, anti-spam,
+      server ACLs, room directory changes, room alias transfers, account
+      deactivation, room shutdown, and more.
+
+      A Synapse module is also available to apply the same rulesets the bot
+      uses across an entire homeserver.
+    '';
+    license = licenses.asl20;
```

Comment:
Draupnir doesn't use the same license as Mjolnir, it's the Cooperative Software License

Reply:
How would I apply this license in Nix? It doesn't look like it's defined in licenses.nix?

Reply:
its also more complicated as the project contains code under 2 separate licences at the same time. External contributors are contributing under Apache 2.0 meanwhile the author of the project post fork contributes under CSL. This is supposed to enable a future option to relicence the whole project back to Apache 2.0 without needing to chase down anyone except the author.

Reply:
This should now be resolved as I've added the CSL license.